pytests/conftest.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import print_function, with_statement, absolute_import
+
+import inspect
+import pytest
+import threading
+import types
+
+from .helpers.session import DebugSession
+
+
+@pytest.fixture
+def daemon():
+    """Provides a factory function for daemon threads. The returned thread is
+    started immediately, and it must not be alive by the time the test returns.
+    """
+
+    daemons = []
+
+    def factory(func):
```

Comment:
>factory(func) [](start = 8, length = 13)

Should this take args?

Reply:
You mean, to pass through  to `Thread` constructor? This is meant for the simplest (but common) case, so I think most of them don't make much sense there. But `name` might be useful, since it is used to identify the thread in stack traces that are dumped if the test times out.

Or better yet, derive name from the function name inside the decorator, but then allow to specify a suffix to distinguish between different daemons produced from the same function.